packages/token-tools/src/css/shadow.ts

```diff
@@ -28,8 +28,9 @@ export function transformShadowLayer(
   const spread = partialAliasOf?.spread
     ? transformAlias(partialAliasOf.spread)
     : transformDimensionValue(value.spread, { transformAlias });
+  const inset = value?.inset ? 'inset' : undefined;
 
-  return [offsetX, offsetY, blur, spread, color].join(' ');
+  return [inset, offsetX, offsetY, blur, spread, color].filter(Boolean).join(' ');
```

Comment:
So a subtle thing, but we actually DON’T want `.filter(Boolean)` since that may filter out any `0`s from the array, which would change the final value! Instead, we should just prepend `inset` if it’s specified. We may not have tests to catch this.

```suggestion
const value = [offsetX, offsetY, blur, spread, color];
if (inset) {
  value.unshift(inset);
}
return value.join(' ');
```

Reply:
Oh, wait—on second thought you’re right; at this point we should have converted all the `0`s to strings (`'0'`), so your logic would be correct! Just double-check to make sure

Reply:
I also increased the explicit check for `true` instead of relying on typescript.

```ts
const inset = value?.inset === true ? 'inset' : undefined;
```

Reply:
And I also went with the explicit filtering. Seems more reliable and less magic, independently of what the number have been transformed to, or not.